src/App.tsx

```diff
@@ -15,48 +15,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
```

Comment:
visibleGoods can be just a common variable there is no need to store it in state